### PR TITLE
Bugfix/matplotlib clang lto

### LINKED ISSUE #23509: Installation issue: py-matplotlib
### Steps to reproduce the issue

```console
$ spack install py-matplotlib%clang
  >> 582    clang-12: error: linker command failed with exit code 1 (use -v to 
            see invocation)
  >> 583    error: command '/scratch/dsjense/spack/lib/spack/env/clang/clang++'
             failed with exit status 1
```

### Information on your system

This fails to build on RHEL7, Ubuntu 18, and Ubuntu 20 with Clang > 10
```
* **Spack:** 0.16.1-2490-5a747b9
* **Python:** 2.7.5
* **Platform:** linux-rhel7-ivybridge
* **Concretizer:** original
```

### Additional information
@adamjstewart 
[This forum post](https://github.com/termux/termux-packages/issues/5787) suggested turning off link-time optimization when using Clang.  I created a branch with a possible fix [here](https://github.com/dsjense/spack/tree/matplotlib-clang-lto).  I'm not sure if this affects all versions of Clang but I definitely had problems with llvm@10,12.

* [spack-build-out.txt]()
* [spack-build-env.txt]()

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [ ] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

### PR DESCRIPTION
**Pull request by ghost**
Should close https://github.com/spack/spack/issues/23509.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-matplotlib/package.py [hunk 1]
```diff
@@ -189,6 +189,8 @@ def configure(self):
                 setup.write('[libs]\n')
                 setup.write('system_freetype = True\n')
                 setup.write('system_qhull = True\n')
+            if self.spec.satisfies('%clang'):
+                setup.write('enable_lto = False\n')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-matplotlib/package.py]: Can you indent this to be inside of the first if-statement? It looks like this option is only available in 3.3+

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-matplotlib/package.py (hunk 1 above)
Comment (ghost) [reply var/spack/repos/builtin/packages/py-matplotlib/package.py]: Good idea, I fixed it.  I'll try building an older version of matplotlib to make sure there isn't some other problem with Spack-built versions of Clang.

Comment (ghost) [reply var/spack/repos/builtin/packages/py-matplotlib/package.py]: py-matplotlib@3.2.2%clang@12 built just fine without any changes to `setup.cfg`.

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
Thanks for the fix, and welcome to Spack!

**Comment by samcom12** [comment]
The same needs to be done when using `Intel-oneapi-compilers`

**Pull request merged**